src/fmu/dataio/providers/objectdata/_base.py

```diff
@@ -62,7 +61,7 @@ class ObjectDataProvider(Provider):
     # result properties; the most important is metadata which IS the 'data' part in
     # the resulting metadata. But other variables needed later are also given
     # as instance properties in addition (for simplicity in other classes/functions)
-    _metadata: dict = field(default_factory=dict)
+    _metadata: AnyContent | UnsetAnyContent | None = field(default=None)
```

Comment:
Don't like needing to leave this as `None` initially, but we are invested in dataclasses at this point